### PR TITLE
SSL Fixes

### PR DESCRIPTION
**Pull request by akamyshanov**
SocketInitiator.DoConnect must clone SocketSettings and then configure the local copy according to the session settings. Otherwise, different section settings are not respected.
Another exception related bug fixed.

### CODE REVIEW

**Comment by akamyshanov** [comment]
Hi! Anything extra I must do for this PR to be merged/rejected?

File: QuickFIXn/Transport/SocketInitiator.cs [hunk 1]
```diff
@@ -54,6 +54,15 @@ public SocketInitiator(IApplication application, IMessageStoreFactory storeFacto
         public static void SocketInitiatorThreadStart(object socketInitiatorThread)
         {
             SocketInitiatorThread t = socketInitiatorThread as SocketInitiatorThread;
+            
+            // just in case
+            if (t == null) return;
+            Action<Exception> hanldeException = ex =>
```
Comment (cbusbey) [inline QuickFIXn/Transport/SocketInitiator.cs]: mind fixing this minor typo? (hanldeException vs handleException...)

**Comment by cbusbey** [comment]
@akamyshanov can you take another look at this pr and sign the cla.  Also I had one minor suggestion on the pr regarding spelling.  Thanks!

**Comment by akamyshanov** [comment]
Hey @cbusbey, fixed the typo.

**Comment by akamyshanov** [comment]
@cbusbey ... and removed the code that contained it with a cherry-pick :) I think that the code is a bit cleaner in 0b071c41beccb258bbeb26ca8eff682f6aeaa333

**Comment by cbusbey** [comment]
@akamyshanov thanks for taking care of that.  One other thing.... it looks like there is a merge conflict with master.  Can you merge or rebase from master so we can get this merged cleanly?  Thanks!

**Comment by akamyshanov** [comment]
@cbusbey done, didn't quite understand what was the conflict until now

**Comment by cbusbey** [comment]
Thanks! :beers:

**Pull request merged**